gems/codetracer-ruby-recorder/ext/native_tracer/src/lib.rs

```diff
@@ -735,93 +783,171 @@ unsafe extern "C" fn event_hook_raw(data: VALUE, arg: *mut rb_trace_arg_t) {
     }
 
     let recorder = &mut *get_recorder(data);
-    if !recorder.active {
+    if !recorder.data.active {
         return;
     }
 
+    if recorder.data.in_event_hook {
+        return;
+    }
+    recorder.data.in_event_hook = true;
+
+    let mut locked_tracer = recorder.tracer.lock().unwrap();
+
     let ev: rb_event_flag_t = rb_tracearg_event_flag(arg);
     let path_val = rb_tracearg_path(arg);
     let line_val = rb_tracearg_lineno(arg);
     let path = rstring_checked_or_empty(path_val);
     let line = rb_num2long(line_val) as i64;
     if should_ignore_path(&path) {
+        recorder.data.in_event_hook = false;
         return;
     }
 
+    let thread_id: u64 = rb_eval_string(c"Thread.current".as_ptr() as *const c_char)
```

Comment:
discussed in discord chat: writing again one question here, sorry for repeating: isn't this slow: to evaluate it as ruby code : it seems we can access the thread_id in other places from `*const rb_internal_thread_event_data_t,` probably here we